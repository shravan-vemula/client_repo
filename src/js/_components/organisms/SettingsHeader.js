import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import BtrackerLogo from "images/Btracker.png";
import { Redirect } from "react-router-dom";

const useStyles = makeStyles((theme) => ({
  root: {
    flexGrow: 1,
  },
  menuButton: {
    marginRight: theme.spacing(2),
    height: 10,
  },
  title: {
    flexGrow: 1,
  },
}));

export default function SettingsHeader() {
  const classes = useStyles();
  const [onClick, setOnClick] = React.useState(false);

  const handleClick = () => {
    setOnClick(true);
  };

  if (onClick === true) {
    return <Redirect to="/home/overview" />;
  }

  return (
    <div className={classes.root}>
      <div id="logo" className="logoBtracker">
        <img
          id="img-click"
          src={BtrackerLogo}
          onClick={handleClick}
          className="BtrackerIcon"
        />
      </div>
    </div>
  );
}
