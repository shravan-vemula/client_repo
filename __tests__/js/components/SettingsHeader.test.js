import React from "react";
import { cleanup } from "@testing-library/react";
import SettingsHeader from "js/_components/organisms/SettingsHeader";
import { render, screen } from "@testing-library/react";
import { BrowserRouter as Router, MemoryRouter } from "react-router-dom";

afterEach(cleanup);

describe("SettingsHeader Component", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<SettingsHeader />);
    expect(wrapper).toMatchSnapshot();
  });
  it("on click check", () => {
    const wrapper = shallow(<SettingsHeader />);
    const img = wrapper.find("#img-click");
    img.simulate("click");
  });
});

describe("HomePageHeader renders correctly", () => {
  test("renders form layout", () => {
    render(
      <Router>
        <SettingsHeader />
      </Router>
    );
    expect(screen.getByRole("img")).toBe;
    screen.debug();
  });
  it("settings header snap shot", () => {
    const wrapper = shallow(
      <MemoryRouter>
        <SettingsHeader />{" "}
      </MemoryRouter>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
