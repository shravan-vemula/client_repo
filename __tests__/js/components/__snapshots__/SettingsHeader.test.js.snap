// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePageHeader renders correctly settings header snap shot 1`] = `ShallowWrapper {}`;

exports[`SettingsHeader Component renders correctly 1`] = `ShallowWrapper {}`;
